languagetool-language-modules/en/src/main/resources/org/languagetool/rules/en/grammar.xml

```diff
@@ -83321,6 +83321,29 @@ The accident victim died from her injuries.
     <!-- Style -->
     <!-- ====================================================================== -->
     <category id="STYLE" name="Style" type="style">
+        <rule id="HAVE_A_SHOWER" name="have a shower -> take a shower" default="temp_off">
+            <antipattern>
+                <token regexp="yes">room|house|home</token>
+                <token></token>
+            </antipattern>
+            <pattern>
+                <marker>
+                    <token inflected="yes">have</token>
+                </marker>
+                <token min="0" postag="RB" />
+                <token postag_regexp="yes" postag="DT|CD" />
+                <token min="0" postag_regexp="yes" postag="JJ|RB"><exception regexp="yes">baby|steam|spacious|big|large|small</exception></token>
+                <token regexp="yes">shower|bath</token>
+            </pattern>
+            <message>The verb <suggestion><match no="1" postag="VB.*">take</match></suggestion> is usually used before '\5'.</message>
```

Comment:
Following Mike's comment, this could be a rule for en-US/grammar.xml. The message could then explain that "having" a shower is not the usual verb to go with `\5` in North America.